app/questionnaire/questionnaire_manager.py

```diff
@@ -83,9 +92,10 @@ def go_to_state(self, item_id):
         node = self.get_state(item_id)
         logger.debug("go to state %s", item_id)
         if node:
+            self._current = node
             logger.debug("truncating to node %s", item_id)
-            if node.next:
-                self._truncate(node.next)
+            # if node.next:
```

Comment:
Removed in a later commit, but missed from this when the commits were cherry-picked. Will fix now